src/spdx2/agent/spdx2utils.php

```diff
@@ -63,7 +63,11 @@ static public function addPrefixOnDemand($license, $spdxValidityChecker = null)
       }
       else
       {
-        return self::$prefix . $license;
+        if(!empty($license)){
+          return self::$prefix . $license;
+        }else{
+          return "NOASSERTION";
```

Comment:
but is there a chance that `addPrefixOnDemand(...` will add a `LicenseREf-` prefix to `NOASSERTION` then?